### PR TITLE
src: replace Promise chains with async/await

### PR DESCRIPTION
**Pull request by cjihrig**
This helps with readability of the code. It also removed some Promises that were not not needed.

### CODE REVIEW

File: src/log.ts [hunk 1]
```diff
@@ -136,36 +136,35 @@ export class Log {
...
+            } else {
+                throw new ApiException<undefined>(
+                    status,
+                    'Error occurred in log request',
+                    undefined,
+                    normalizeResponseHeaders(response),
+                );
+            }
+        } catch (err: any) {
+            throw new ApiException<undefined>(err, 'Error occurred in log request', undefined, err);
```
Comment (cjihrig) [inline src/log.ts:166]: Note - this line is ported from the previous `catch()` call, but I don't think passing `err` as the first parameter is actually correct. `catch()` typed `err` as `any`. When I changed to a `try...catch`, that typing was not there and TypeScript complained that the first argument should be a numeric code (probably similar to how `src/metrics.ts` does it).

Comment (brendandburns) [reply src/log.ts:166]: Yeah, I'm on board with switching this to match metric.ts

**Comment by brendandburns** [comment]
This basically looks good to me, one response to your query, but happy to merge as-is if you prefer,

**Comment by cjihrig** [comment]
I've updated the `catch` block to work like the one in the metrics file.

**Comment by mstruebing** [comment]
/lgtm
/approve

Thanks 🚀

**Pull request merged**